lldb/source/Plugins/SymbolFile/DWARF/SymbolFileDWARFDebugMap.cpp

```diff
@@ -246,6 +247,8 @@ llvm::StringRef SymbolFileDWARFDebugMap::GetPluginDescriptionStatic() {
 }
 
 SymbolFile *SymbolFileDWARFDebugMap::CreateInstance(ObjectFileSP objfile_sp) {
+  if (objfile_sp->GetPluginName() != ObjectFileMachO::GetPluginNameStatic())
+    return nullptr;
```

Comment:
Who is calling this? I'm wondering if the check might make more sense at the call site?